### PR TITLE
Store and apply `overrides` per-compiler

### LINKED ISSUE #5755: [BUG]: Unsupported compiler overrides are kept when switching compilers, leading to confusing UI issues
### Describe the bug

Unsupported overrides are kept when switching compilers which leads to a confusing difference between what the menu shows, the overrides button's active state implies, and what the compiler actually ends up using.

### Steps to reproduce

1. Enable override A for compiler 1
2. Switch to compiler 2, which does not support override A
3. The overrides button will be green to show there is an override, but the compiler vetoes it and doesn't pass it to the command line.
4. If you were to open the overrides menu here, the unsupported override would be deleted.
5. Switch back to compiler 1
6. If you did not open the menu while on compiler 2, the override is now back, as compiler 1 supports it. If you did open the menu, it's gone and you would have to put it back.

In the reproducer link, I:
* Added an "action" override for TableGen.
* Switched to the gcc compiler for C.
* You will see that it claims to have an active override for `action`, but in fact, gcc doesn't support that override.
* Opening the menu will clear `action`.

### Expected behavior

I have no idea what the expected behaviour here was intended to be, but I can think of 2 models that might work:
1. Switching to a new compiler removes all unsupported overrides. When you go back to the first compiler you must add them again (draft PR: https://github.com/compiler-explorer/compiler-explorer/pull/5747)
2. Unsupported overrides are kept when switching compilers, simply hidden from the UI, and do not cause the overrides button to be green (draft PR: https://github.com/compiler-explorer/compiler-explorer/pull/5744)

Option 1 is the simplest to implement, option 2 would mean you can "carry" an option between compilers at least temporarily.

Maybe you compile for target pentiumm with gcc, and switch to MSVC quickly, then go back. It might be nice to have kept the target option so you don't have to add it. But again, I'm not sure what the original intent here is.

The libraries menu is a bit different because it lists only enabled libraries that are compatible with the current compiler. I am not sure if it removes unsupported libraries while doing this though.

### Reproduction link

https://godbolt.org/z/aPPeP3bzs

### Screenshots

_No response_

### Operating System

_No response_

### Browser version

_No response_

### PR DESCRIPTION
**Pull request by OfekShilon**
Fixes #5755 and maybe others.

### CODE REVIEW

**Comment by OfekShilon** [comment]
There are 2 languages with default overrides: rust and tablegen. One of the effects of this PR is that the overrides button initially *has* green highlight for these languages.   I can change it back, but I tend to think it makes more UX sense this way.

File: lib/compiler-finder.ts [hunk 1]
```diff
@@ -345,6 +345,7 @@ export class CompilerFinder {
                 preamble: props<string>('licensePreamble'),
             },
             possibleOverrides: [],
+            activeOverrides: [],
```
Comment (partouf) [inline lib/compiler-finder.ts:348]: This is too confusing of a name, once this is transmitted to the frontend this object should not ever change. But then the Default overrides are also already in `possibleOverrides`, so I'm not sure what's the point of this.

File: static/panes/compiler.ts [hunk 2]
```diff
@@ -316,6 +316,10 @@ export class Compiler extends MonacoPane<monaco.editor.IStandaloneCodeEditor, Co
 
         this.revealJumpStack = [];
 
+        if (this.compiler) {
+            this.compiler.activeOverrides = state.overrides;
```
Comment (partouf) [inline static/panes/compiler.ts:320]: Best to store this somewhere else like the global https://github.com/compiler-explorer/compiler-explorer/blob/main/static/global.ts#L30 , or in local storage if you want it to persist beyond refreshes

File: lib/compiler-finder.ts (hunk 1 above)
Comment (OfekShilon) [reply lib/compiler-finder.ts:348]: Not sure I understand this.
IIUC this (compiler-finder stuff) runs only on instance startup. Why shouldn't `activeOverrides` change afterwards?
And what better name would you suggest?

Comment (partouf) [reply lib/compiler-finder.ts:348]: Because it's supposed to represent the static configuration on the backend for everyone, not an individualized state

Comment (OfekShilon) [reply lib/compiler-finder.ts:348]: Got it. Will take another stab at this, probably next weekend

**Comment by OfekShilon** [comment]
Closing, hopefully will re-attack this some day in the future

**Pull request closed without merging**